lightning/src/chain/keysinterface.rs

```diff
@@ -927,8 +935,9 @@ impl KeysManager {
 	/// output to the given change destination (if sufficient change value remains). The
 	/// transaction will have a feerate, at least, of the given value.
 	///
-	/// Returns `Err(())` if the output value is greater than the input value minus required fee or
-	/// if a descriptor was duplicated.
+	/// Returns `Err(())` if the output value is greater than the input value minus required fee,
+	/// if a descriptor was duplicated, or if p2wpkh script does not 
+	/// match the descriptor output script.
```

Comment:
Ah, we should probably update the body to stop unwrap'ing the other spend calls, then update the doc here to just say something like the generic text I pasted above.